### PR TITLE
Support llava inference example

### PR DESCRIPTION
**Pull request by jokmingwong**
Considering 2 special cases and fix bugs: 
1. boolean computation on cumsum, eq, mul, sum 
2. negative index value on CrossEntropyLoss

### CODE REVIEW

File: src/flag_gems/ops/eq.py [hunk 1]
```diff
@@ -13,7 +13,14 @@ def eq_func(x, y):
 
 def eq(A, B):
     logging.debug("GEMS EQ")
-    O = eq_func(A, B)
+    if A.dim() == 0 and B.dim()==0:
+        O = eq_func_scalar(A.item(), B.item())
+    elif A.dim() == 0:
```
Comment (StrongSpoon) [inline src/flag_gems/ops/eq.py]: can't 0-dim tensor broadcast to high-dimension?

**Review by StrongSpoon — DISMISSED** [review]

File: src/flag_gems/ops/eq.py (hunk 1 above)
Comment (StrongSpoon) [reply src/flag_gems/ops/eq.py]: pointwise_dynamic can deal with 0-dim tensor. while the problem here is that equal may receive a cuda tensor and a cpu tensor. the solution is cast cpu tensor to the same device as cuda tensor.

File: src/flag_gems/ops/eq.py [hunk 2]
```diff
@@ -13,6 +13,11 @@ def eq_func(x, y):
 
 
 def eq(A, B):
+    if A.device != B.device:
+        if A.device.type == "cuda":
+            B = B.to(A.device)
+        else:
+            A = A.to(B.device)
     logging.debug("GEMS EQ")
     return eq_func(A, B)
```
Comment (Bowen12992) [inline src/flag_gems/ops/eq.py:22]: should we add device guard here？

Comment (StrongSpoon) [reply src/flag_gems/ops/eq.py:22]: device guard is inside code generation

**Review by Bowen12992 — APPROVED** [review]
LGTM, we can add model llava test into `.github/workflows/python-test.yaml`

**Pull request merged**